### ISSUE TITLE
Toward Concrete Execution

### ISSUE THREAD

**Issue by douggard** [trailofbits/manticore#307]
### Summary 
Looking into executing using the Linux platform (as opposed to SLinux), I ran into a few issues and am looking for feedback on how they should be fixed before submitting a PR. The changes are minimal.

#### _symbols

`core/executor.py` - `Memory64` does not have `_symbols`. In the state initialization [here](https://github.com/trailofbits/manticore/blob/master/manticore/core/executor.py#L305), `smem` is defined as `len(state.platform.current.memory._symbols)`, since `Memory64` does not have `_symbols` this causes an error. Options:

1. Add dummy `_symbols` to `Memory64`.

2. Do a check in `executor.py` and only include `smem` if it is the appropriate memory type.

#### asserts

`core/state.py` - There are two asserts [here](https://github.com/trailofbits/manticore/blob/master/manticore/core/state.py#L93) that cause a no attribute error. `assert self.mem.constraints is self.constraints` Options:

1. Remove the asserts.

2. Add dummy `constraints`to `mem` and `platform`.

#### sys_mmap

There is a `sys_mmap` defined in SLinux [here](https://github.com/trailofbits/manticore/blob/master/manticore/platforms/linux.py#L1960), I don't see anything specifically symbolic in it, but I didn't follow the functions it called. I'll explore this further but the simplest solution is to move the non-symbolic syscalls to Linux.

#### sys_open

`sys_open` references self.symbolic_files in Linux [here](https://github.com/trailofbits/manticore/blob/master/manticore/platforms/linux.py#L1185). I think this should define a non-symbolic one in Linux then redefine under SLinux, this causes some code duplication though where the symbolic version might fix a concrete issue but that wouldn't propagate back to the Linux class. Thoughts?

#### sys_lseek

There is a corner case here that I ran into on my test application. Python integers max out at `2**63-1` then become a long. There creates a situation if you have a 64 bit negative number, passing it to file seek causes issues since its out of the Python "int" range. A simple solution is to `import ctypes` and use:

```python
if offset > sys.maxsize and sys.maxsize == (2**63 - 1):
    offset = ctypes.c_long(offset).value
```

This will convert it to the appropriate negative number. Additionally, `whence` is not passed to seek. That change is also simple, changing the call to `self.files[fd].seek(offset, whence)`.

**Comment by mossberg** [comment]
- moving sys_mmap to Linux sounds great, i don't think it belongs in SLinux anyway
- for sys_open what if we abstract these lines (https://github.com/trailofbits/manticore/blob/master/manticore/platforms/linux.py#L1185-L1191) into a function that returns `f`? in `Linux` it will simply call the `File` constructor, but in SLinux, it will basically be what those lines are now

**Comment by douggard** [comment]
Bugs fixed in #337.
